Tests/AblyChatTests/DefaultRoomPresenceTests.swift

```diff
@@ -0,0 +1,465 @@
+
+        // Given: A DefaultPresence with DefaultFeatureChannel and MockRoomLifecycleContributor
+        let realtimePresence = MockRealtimePresence(members: ["client1"].map { .init(clientId: $0) })
+        let channel = MockRealtimeChannel(name: "basketball::$chat::$chatMessages", mockPresence: realtimePresence)
+        let contributor = createMockContributor(feature: .presence, attachBehavior: .complete(.failure(error)))
+        let featureChannel = DefaultFeatureChannel(channel: channel, contributor: contributor, roomLifecycleManager: lifecycleManager)
+        let defaultPresence = await DefaultPresence(featureChannel: featureChannel, roomID: "basketball", clientID: "mockClientId", logger: TestLogger())
+
+        // When: The room is in the attaching state
+        let roomStatusSubscription = await lifecycleManager.onRoomStatusChange(bufferingPolicy: .unbounded)
+
+        async let _ = lifecycleManager.performAttachOperation()
+
+        // Wait for room to become ATTACHING
+        _ = try #require(await roomStatusSubscription.attachingElements().first { _ in true })
+
+        // TODO: avoids compiler crash (https://github.com/ably/ably-chat-swift/issues/233), revert once Xcode 16.3 released
+        let doIt = {
+            try await defaultPresence.enter()
+        }
+        await #expect {
+            try await doIt()
+        } throws: { error in
+            isChatError(error, withCodeAndStatusCode: .variableStatusCode(.roomInInvalidState, statusCode: 500), cause: attachError)
+        }
+
+        // Then: The manager were waiting for its room status to change from attaching
+        await #expect(lifecycleManager.waitCallCount == 1)
+    }
+
+    // @spec CHA-PR3h
+    @Test
+    func failToEnterPresenceWhenRoomInInvalidState() async throws {
+        // Given
+        let realtimePresence = MockRealtimePresence(members: ["client1"].map { .init(clientId: $0) })
+        let channel = MockRealtimeChannel(name: "basketball::$chat::$chatMessages", mockPresence: realtimePresence)
+        let error = ARTErrorInfo(chatError: .presenceOperationRequiresRoomAttach(feature: .presence))
+        let featureChannel = MockFeatureChannel(channel: channel, resultOfWaitToBeAblePerformPresenceOperations: .failure(error))
+        let defaultPresence = await DefaultPresence(featureChannel: featureChannel, roomID: "basketball", clientID: "mockClientId", logger: TestLogger())
+
+        // Then
+        // TODO: avoids compiler crash (https://github.com/ably/ably-chat-swift/issues/233), revert once Xcode 16.3 released
+        let doIt = {
+            _ = try await defaultPresence.enter()
+        }
+        await #expect {
+            try await doIt()
+        } throws: { error in
+            isChatError(error, withCodeAndStatusCode: .variableStatusCode(.roomInInvalidState, statusCode: 400), partialDescription: "attach")
```

Comment:
Why can't we test for the full error message? Just checking for the word "attach" doesn't seem to tell us a lot.

Reply:
done